README.md

```diff
@@ -46,12 +46,274 @@ setting up a database inside an enclave is very well possible. This sample shows
+a string to the enclave, the enclave will reverse it and send it back to the client.
 
-    conclaveRepo=add_path_to_repo
+#Step 1: Prerequisites
+You need [JDK](https://www.openlogic.com/openjdk-downloads) 8 or 11 installed in order to build and run the app.<br />
+Grab a copy of Conclave-SDK from [conclave.net](https://www.conclave.net/get-conclave/)
+We will use the conclave init tool for bootstrapping Conclave projects, reducing the amount of boilerplate you need to write.
+</br>You will find conclave-init.jar in the tools directory of the Conclave SDK.
+
+# Step 2: Create the project template using the conclave-init tool
+
+To create a new Conclave project, run the following command:
+
+    java -jar /path/to/conclave-sdk/tools/conclave-init.jar 
+    --package com.megacorp 
+    --enclave-class-name ReverseEnclave 
+    --target ./reverse-conclave-app
+
+A new project would have been created in the reverse-conclave-app directory, as specified by --target ./reverse-conclave-app.
+</br>This generates three modules
+
+1. The enclave module contains the ReverseEnclave class. Your enclave code will go here.
+2. The host directory has been created, too.
+   It contains only a build.gradle file, since that's all that's required for the [Conclave web host](https://docs.conclave.net/conclave-web-host.html).
+3. The client directory contains a basic client which can interact with the web host.
+
+To understand more about the different files generated by this tool jump to the below section named Configurations generated by the conclave-init tool.
+
+# Step 3: Add the Enclave code
+
+Copy paste the below code to receiveMail method in SimpleEnclave class. This logic reverses the string received by the client and sends it back to the client.
+
+      String inputString = new String(mail.getBodyAsBytes());
+      StringBuilder outputString = new StringBuilder();
+      outputString.append(inputString);
+      outputString.reverse();
+      byte[] response = postOffice(mail).encryptMail(outputString.toString().getBytes());
+      postMail(response, routingHint);
+
+# Step 4: Run this sample
+
+There are different [modes](https://docs.conclave.net/enclave-modes.html) in which you can run your Conclave application.
+For this tutorial we will run the conclave application in mock mode. To learn more about the different modes, please
+read the below section named Enclave Modes.
+
+1. Build the host
+
+        ./gradlew clean host:shadowJar -PenclaveMode=mock
+
+2. Build the client
+
+        ./gradlew client:shadowJar
+
+3. Run the host
+
+        java -jar host/build/libs/host-mock.jar
+
+5. Run the client
+
+        java -jar client/build/libs/client-all.jar "S:0000000000000000000000000000000000000000000000000000000000000000 PROD:1 SEC:INSECURE" ReverseMe
+
+Using above 4 steps you should be able to have a conclave application running on your system.
```

Comment:
Maybe add a final line such as: "Now you have developed and tested your enclave, you can take a final step, to achieve the privacy promise of Conclave. To do this, you can run the enclave in 'RELEASE' mode. See here.

Reply:
done